lib/internal/event_target.js

```diff
@@ -564,14 +565,13 @@ function isEventTarget(obj) {
 }
 
 function addCatch(that, promise, event) {
-  const then = promise.then;
-  if (typeof then === 'function') {
-    then.call(promise, undefined, function(err) {
-      // The callback is called with nextTick to avoid a follow-up
-      // rejection from this promise.
-      process.nextTick(emitUnhandledRejectionOrErr, that, err, event);
-    });
-  }
+  if (!SafeThenable)
+    SafeThenable = require('internal/per_context/safethenable');
+  new SafeThenable(promise).catch(function(err) {
```

Comment:
What advantage does this have over `{ await thenable } catch (err) { ...`?

Reply:
This is not awaiting the _thenable_  at all – it attaches a catch handler in case the event handler returns a _thenable_. I'm pretty sure that's in the spec, although I haven't checked.